nifi-nar-bundles/nifi-cdc/nifi-cdc-postgresql-bundle/nifi-cdc-postgresql-processors/src/test/java/org/apache/nifi/cdc/postgresql/processors/ITCaptureChangePostgreSQL.java

```diff
@@ -0,0 +1,1113 @@
+                runner.assertValid();
+
+                runner.setProperty(CaptureChangePostgreSQL.DRIVER_NAME, "org.postgrez.Driver"); // Incorrect driver name
+                try {
+                        runner.run();
+                } catch (AssertionError ae) {
+                        Throwable pe = ae.getCause();
+                        assertTrue(pe instanceof ProcessException);
+                        Throwable rue = pe.getCause();
+                        assertTrue(rue instanceof RuntimeException);
+                        assertEquals(
+                                        "Failed to register JDBC driver. Ensure PostgreSQL Driver Location(s) and PostgreSQL Driver Class Name "
+                                                        + "are configured correctly. org.apache.nifi.reporting.InitializationException: Can't load Database Driver",
+                                        rue.getMessage());
+                        Throwable ine = rue.getCause();
+                        assertTrue(ine instanceof InitializationException);
+                        assertEquals("Can't load Database Driver", ine.getMessage());
+                }
+                runner.setProperty(CaptureChangePostgreSQL.DRIVER_NAME, DRIVER_NAME);
+        }
+
+        @Test
+        public void test03ConnectionFailures() {
+                runner.setProperty(CaptureChangePostgreSQL.DRIVER_LOCATION, DRIVER_LOCATION);
+                runner.setProperty(CaptureChangePostgreSQL.DRIVER_NAME, DRIVER_NAME);
+                runner.setProperty(CaptureChangePostgreSQL.HOST, HOST);
+                runner.setProperty(CaptureChangePostgreSQL.PORT, PORT);
+                runner.setProperty(CaptureChangePostgreSQL.DATABASE, DATABASE);
+                runner.setProperty(CaptureChangePostgreSQL.USERNAME, USERNAME);
+                runner.setProperty(CaptureChangePostgreSQL.PASSWORD, PASSWORD);
+                runner.setProperty(CaptureChangePostgreSQL.CONNECTION_TIMEOUT, CONNECTION_TIMEOUT);
+                runner.setProperty(CaptureChangePostgreSQL.PUBLICATION, PUBLICATION);
+                runner.setProperty(CaptureChangePostgreSQL.REPLICATION_SLOT, REPLICATION_SLOT);
+
+                runner.setProperty(CaptureChangePostgreSQL.HOST, "192.168.56.199"); // Incorrect IP
+                try {
+                        runner.run();
+                } catch (AssertionError ae) {
+                        Throwable pe = ae.getCause();
+                        assertTrue(pe instanceof ProcessException);
+                        Throwable ioe = pe.getCause();
+                        assertTrue(ioe instanceof IOException);
+                        assertEquals("Error creating SQL connection to specified host and port", ioe.getMessage());
+                        Throwable psqle = ioe.getCause();
+                        assertTrue(psqle instanceof PSQLException);
+                }
```

Comment:
This type of evaluation should be changed to use JUnit `assertThrows()` with the expected exception as the first parameter.

Reply:
Other processors did the same. I used them as an example. But, no problem, I'll make this improvement in the tests.

Reply:
There is a mix of approaches right now, newer tests leverage the `assertThrows()` method, while older tests still have the try-catch approach.  Thanks for working through the changes!

Reply:
Done!